modular_skyrat/modules/synths/code/bodyparts/power_cord.dm

```diff
@@ -0,0 +1,83 @@
+#define SYNTH_CHARGE_MAX 150
+#define SYNTH_CHARGE_MIN 50
+#define SYNTH_CHARGE_PER_NUTRITION 10
+#define SYNTH_CHARGE_DELAY_PER_100 10
+#define SYNTH_DRAW_NUTRITION_BUFFER 30
+
+/obj/item/organ/internal/cyberimp/arm/power_cord
+	name = "power cord implant"
+	desc = "An internal power cord. Useful if you run on elecricity. Not so much otherwise."
+	contents = newlist(/obj/item/apc_powercord)
+	zone = "l_arm"
+
+/obj/item/apc_powercord
+	name = "power cord"
+	desc = "An internal power cord. Useful if you run on electricity. Not so much otherwise."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "wire1"
+
+/obj/item/apc_powercord/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!istype(target, /obj/machinery/power/apc) || !ishuman(user) || !proximity_flag)
+		return ..()
+	user.changeNext_move(CLICK_CD_MELEE)
+	var/obj/machinery/power/apc/target_apc = target
+	var/mob/living/carbon/human/ipc = user
+	var/obj/item/organ/internal/stomach/synth/cell = ipc.internal_organs_slot[ORGAN_SLOT_STOMACH]
+	if(!cell)
+		to_chat(ipc, span_warning("You try to siphon energy from the [target_apc], but you have no stomach! How are you still standing?"))
+		return
+	if(!istype(cell))
+		to_chat(ipc, span_warning("You plug into the APC, but nothing happens! It seems you don't have a cell to charge!"))
+		return
+
+	if(target_apc.cell && target_apc.cell.charge > 0)
+		if(ipc.nutrition >= NUTRITION_LEVEL_WELL_FED)
+			to_chat(user, span_warning("You are already fully charged!"))
+			return
+		else
+			powerdraw_loop(target_apc, ipc)
+			return
+
+	to_chat(user, span_warning("There is no charge to draw from that APC."))
```

Comment:
I feel like this to_chat at the end should be inside of that `if()` instead, and the other two should be at the end instead, to reduce the overall indentation in that proc.